packages/contracts/contracts/optimistic-ethereum/chain/CanonicalTransactionChain.sol

```diff
@@ -79,8 +83,8 @@ contract CanonicalTransactionChain {
             "Must process older SafetyQueue batches first to enforce timestamp monotonicity"
         );
 
-        _appendQueueBatch(l1ToL2Header, true);
         l1ToL2Queue.dequeue();
+        _appendQueueBatch(l1ToL2Header, true);
```

Comment:
Hmm does this ordering matter/why does this ordering matter? Just want to understand if there are security implications

Reply:
You're right, it shouldn't. I'm just used to making calls after changing state to prevent reentrancy concerns, but this isn't an external call or anything, so we're fine.